ts/defs.bzl

```diff
@@ -153,6 +154,10 @@ def ts_project(
             See https://www.typescriptlang.org/docs/handbook/compiler-options.html#compiler-options
             Typically useful arguments for debugging are `--listFiles` and `--listEmittedFiles`.
 
+        async_typecheck: Whether to type-check asynchronously as a separate bazel action.
```

Comment:
TODO: names:

* emphasize type-checking is a separate action: (parallel|async|defer|non_blocking|independent)(_typecheck)?
* emphasize transpiling is faster and doesn't wait for deps: (untyped|nocheck|parallel)(_transpile)?
* do it automatically instead of opt-in when isolated_declarations && is_ts56

Reply:
@dzbarsky [says](https://github.com/aspect-build/rules_ts/pull/698#discussion_r1769024468)
> For us we did a single flag enforce_isolated_declarations that enabled this mode + isoDecls in the tsconfig. Do you think there's any case where someone would set isoDecls in tsconfigs but not want to enable this?

One reason would be when not on typescript >=5.6. Otherwise I'm not sure... would it ever be slower running tsc twice like this?

Overall I'm actually in favour of the 3rd option I listed which I think aligns with yours where we'd do this automatically when `isolatedDeclarations` is set, and on typescript >=5.6.

Reply:
My vote would be to do it automatically, and if anyone complains, provide an opt-out mechanism for them. I think that's default people will want

Reply:
Then we just need the typescript >=5.6 check which I think we can do the exact same as `is_typescript_5_or_greater` today?

Reply:
yes, the condition is slightly more complex but I agree with your assessment

Reply:
@jbedard and I chatted about flipping the default of `isolated_typecheck` to True if ts>5.6, however, it depends on the user having set `isolatedDeclarations` to true in their tsconfig, which we can't know at analysis time, so there is no way to set a reasonable default. Punting on the automatic opt-in for now.

Reply:
Note we decided on `isolated_typecheck` as the name for this param instead of the `async_typecheck` I originally had.